iexec-sms-library/src/main/java/com/iexec/sms/api/SmsService.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.api;
+
+import com.iexec.common.chain.WorkerpoolAuthorization;
+import com.iexec.common.tee.TeeWorkflowSharedConfiguration;
+import com.iexec.common.utils.BytesUtils;
+import feign.FeignException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Slf4j
+public class SmsService {
+
+    private final SmsClient smsClient;
+
+    private TeeWorkflowSharedConfiguration teeWorkflowConfiguration;
+
+    public SmsService(SmsClient smsClient) {
```

Comment:
If we start exposing an `SmsService`, we shall hide the `SmsClient`, since the connection to the SMS is handled by the service itself.

We might update the constructor to these
```
public SmsService(String smsUrl)

public SmsService(String smsUr, Logger.level levell)
```

Reply:
The question is where do you put your cursor, I mean what do you consider as your client (for consumers).
I'm currently not sure to see the added value of an SmsService since we already have an SmsClient.
Here is what I have in mind for client usage:
```
AmazonS3 s3client = AmazonS3ClientBuilder
  .standard()
  .withCredentials(new AWSStaticCredentialsProvider(credentials))
  .withRegion(Regions.US_EAST_2)
  .build();
```
```
try {
    s3client.deleteBucket("baeldung-bucket-test2");
} catch (AmazonServiceException e) {
    System.err.println("e.getErrorMessage());
    return;
}
```
https://www.baeldung.com/aws-s3-java#creating-client-connection

Reply:
Are you sure you wish to take as reference a tutorial explaining how to use a library to take a decision on software architecture ?

For me the interest (or not) of a service class lies with the following question.
Do we need a common way to handle errors ? Do we want to provide a single and fully validated implementation ?
If yes, the SmsService class has added value.
If no, this PR needs to be dropped and the iexec-sms-library may have no meaning.
We may even be better with defining feign clients in each application by defining only the relevant endpoints as it was.

Reply:
Just put the sample tutorial for simplification purposes, the implementation to dig into is here: https://github.com/aws/aws-sdk-java/tree/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3 (AmazonS3ClientBuilder.java, ..)

Reply:
Let me be clear, the SmsService is totally meaningful if it is the entry point for consumers. With current PR, the consumer has mutiple potential entry points (Direct access with SmsClient or access with SmsService).

Reply:
constructor added in 4e756d3